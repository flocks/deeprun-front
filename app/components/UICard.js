import React, { PropTypes } from 'react';
import cx from 'classnames';
import styles from '../styles/card.scss';

const UICard = ({ ...props }) => {
    const handleClick = () => {
        if (!props.disabled) {
            props.onClick();
        }
    };

    const suit = props.range.charAt(1);

    let className = cx(styles.card, styles[suit], {
        [styles.selected]: props.selected,
        [styles.disabled]: props.disabled
    });

    return (
        <div
            className={ className }
            onClick={handleClick}
        >
            <div>{props.range.charAt(0)}</div>
        </div>
    );
};

UICard.defaultProps = {
    range: '',
    onClick: function onClick() {},
    disabled: false,
    selected: false
};

UICard.propTypes = {
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    range: PropTypes.string,
    selected: PropTypes.bool,
    style: PropTypes.object
};


export default UICard;
