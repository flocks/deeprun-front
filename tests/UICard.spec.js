import React from 'react';
import renderer from 'react-test-renderer';
import UICard from '../app/components/UICard';

describe('<UICard />', () => {
    it('Empty cards should render properly', () => {
        const component = renderer.create(
            <UICard />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Empty card should render properly', () => {
        const component = renderer.create(
            <UICard />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Disabled card should render properly', () => {
        const component = renderer.create(
            <UICard card={{rank: '3', suit: 'spade'}} disabled />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Selected cards should render properly', () => {
        const spade = renderer.create(
            <UICard card={{rank: '8', suit: 'spade'}} selected />
        );

        let tree = spade.toJSON();
        expect(tree).toMatchSnapshot();

        const heart = renderer.create(
            <UICard card={{rank: '8', suit: 'heart'}}  selected />
        );

        tree = heart.toJSON();
        expect(tree).toMatchSnapshot();

        const diamond = renderer.create(
            <UICard card={{rank: '8', suit: 'diamond'}} selected />
        );

        tree = diamond.toJSON();
        expect(tree).toMatchSnapshot();

        const club = renderer.create(
            <UICard card={{rank: '8', suit: 'club'}} selected />
        );

        tree = club.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('cards should render properly', () => {
        const spade = renderer.create(
            <UICard card={{rank: '8', suit: 'spade'}} />
        );

        let tree = spade.toJSON();
        expect(tree).toMatchSnapshot();

        const heart = renderer.create(
            <UICard card={{rank: '8', suit: 'heart'}} />
        );

        tree = heart.toJSON();
        expect(tree).toMatchSnapshot();

        const diamond = renderer.create(
            <UICard card={{rank: '8', suit: 'diamond'}} />
        );

        tree = diamond.toJSON();
        expect(tree).toMatchSnapshot();

        const club = renderer.create(
            <UICard card={{rank: '8', suit: 'club'}}/>
        );

        tree = club.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
