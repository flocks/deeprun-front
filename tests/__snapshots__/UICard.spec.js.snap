// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UICard /> Disabled card should render properly 1`] = `
<div
  className="undefined"
  onClick={[Function]}
>
  <div>
    
  </div>
</div>
`;

exports[`<UICard /> Empty card should render properly 1`] = `
<div
  className=""
  onClick={[Function]}
>
  <div>
    
  </div>
</div>
`;

exports[`<UICard /> Empty cards should render properly 1`] = `
<div
  className=""
  onClick={[Function]}
>
  <div>
    
  </div>
</div>
`;

exports[`<UICard /> Selected cards should render properly 1`] = `
<div
  className=""
  onClick={[Function]}
>
  <div>
    
  </div>
</div>
`;

exports[`<UICard /> Selected cards should render properly 2`] = `
<div
  className=""
  onClick={[Function]}
>
  <div>
    
  </div>
</div>
`;

exports[`<UICard /> Selected cards should render properly 3`] = `
<div
  className=""
  onClick={[Function]}
>
  <div>
    
  </div>
</div>
`;

exports[`<UICard /> Selected cards should render properly 4`] = `
<div
  className=""
  onClick={[Function]}
>
  <div>
    
  </div>
</div>
`;

exports[`<UICard /> cards should render properly 1`] = `
<div
  className=""
  onClick={[Function]}
>
  <div>
    
  </div>
</div>
`;

exports[`<UICard /> cards should render properly 2`] = `
<div
  className=""
  onClick={[Function]}
>
  <div>
    
  </div>
</div>
`;

exports[`<UICard /> cards should render properly 3`] = `
<div
  className=""
  onClick={[Function]}
>
  <div>
    
  </div>
</div>
`;

exports[`<UICard /> cards should render properly 4`] = `
<div
  className=""
  onClick={[Function]}
>
  <div>
    
  </div>
</div>
`;
